Either we mark the package broken on aarch64 or remove the x86 only flags:


```
       > configuring
       > no configure script, doing nothing
       > building
       > build flags: SHELL=/nix/store/52wfs5n7vx8yyq5d7l0r734p78hn0wmd-bash-5.1-p16/bin/bash PREFIX=\$\(out\)
       > gcc -Wall -O3 -I src -msse2 -mfpmath=sse -std=c11 -D LIBHL_EXPORTS -m64 -fPIC -pthread -fno-omit-frame-pointer -o include/pcre/pcre_chartables.o -c include/pcre/pcre_chartables.c
       > gcc: error: unrecognized command line option '-msse2'
       > gcc: error: unrecognized command line option '-mfpmath=sse'
       > gcc: error: unrecognized command line option '-m64'
       > make: *** [Makefile:143: include/pcre/pcre_chartables.o] Error 1
```

--- a/pkgs/development/interpreters/hashlink/default.nix
+++ b/pkgs/development/interpreters/hashlink/default.nix
@@ -1,0 +1,53 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, libGL
+, libGLU
+, libpng
+, libjpeg_turbo
+, libuv
+, libvorbis
+, mbedtls
+, openal
+, pcre
+, SDL2
+, sqlite
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hashlink";
+  version = "1.12";
+
+  src = fetchFromGitHub {
+    owner = "HaxeFoundation";
+    repo = "hashlink";
+    rev = version;
+    sha256 = "AiUGhTxz4Pkrks4oE+SAuAQPMuC5T2B6jo3Jd3sNrkQ=";
+  };
+
+  patches = [ ./hashlink.patch ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  buildInputs = [
+    libGL
+    libGLU
+    libjpeg_turbo
+    libpng
+    libuv
+    libvorbis
+    mbedtls
+    openal
+    pcre
+    SDL2
+    sqlite
+  ];
+
+  meta = with lib; {
+    description = "A virtual machine for Haxe";
+    homepage = "https://hashlink.haxe.org/";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ iblech locallycompact ];
+  };
+}

--- a/pkgs/development/interpreters/hashlink/hashlink.patch
+++ b/pkgs/development/interpreters/hashlink/hashlink.patch
@@ -1,0 +1,8 @@
+*** a/Makefile  1970-01-01 01:00:01.000000000 +0100
+--- b/Makefile  2022-06-21 23:36:10.023460654 +0200
+*************** endif
+*** 109,110 ****
+--- 109,111 ----
+  LIBOPENAL = -lopenal
++ LIBOPENGL = -lGL
+  RELEASE_NAME = linux

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -14421,6 +14421,8 @@ with pkgs;
   hadoop3 = hadoop_3_3;
   hadoop = hadoop3;
 
+  hashlink = callPackage ../development/interpreters/hashlink { };
+
   io = callPackage ../development/interpreters/io { };
 
   ivy = callPackage ../development/interpreters/ivy { };

